### PR TITLE
Add C implementation for `@stdlib/math/base/special/log1pexp`

### LINKED ISSUE #760: [RFC]: Add C implementation for `@stdlib/math/base/special/log1pexp`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/log1pexp`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/log1pexp).

```c
double stdlib_base_log1pexp( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #760.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a review.

File: lib/node_modules/@stdlib/math/base/special/log1pexp/README.md [hunk 1]
```diff
@@ -77,6 +77,97 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+<!-- C API usage examples. -->
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/log1pexp.h"
+#include <math.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/log1pexp/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/log1pexp/README.md [hunk 2]
```diff
@@ -77,6 +77,97 @@ for ( i = 0; i < x.length; i++ ) {
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x;
+    double v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        x = ( ( ( double ) rand() / ( double ) RAND_MAX ) * 20.0 ) - 10.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/log1pexp/README.md]:
```suggestion
        x = ( ( (double)rand() / (double)RAND_MAX ) * 20.0 ) - 10.0;
```

File: lib/node_modules/@stdlib/math/base/special/log1pexp/examples/c/example.c [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/log1pexp.h"
+#include <math.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/log1pexp/examples/c/example.c]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/log1pexp/examples/c/example.c [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x;
+    double v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        x = ( ( ( double ) rand() / ( double ) RAND_MAX ) * 20.0 ) - 10.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/log1pexp/examples/c/example.c]:
```suggestion
        x = ( ( (double)rand() / (double)RAND_MAX ) * 20.0 ) - 10.0;
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Comment by Pranavchiku** [comment]
Apologies for messing up with the headers, and braces convention, will try not to repeat those.

**Pull request merged**